### PR TITLE
Integrate Eth client and EVM-based subnets

### PR DESCRIPTION
**Pull request by adlrocha**
Superseded by: https://github.com/consensus-shipyard/ipc-agent/pull/221

### CODE REVIEW

File: src/manager/evm.rs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+use crate::lotus::message::mpool::MpoolPushMessage;
+use crate::lotus::message::state::StateWaitMsgResponse;
+use crate::lotus::message::wallet::WalletKeyType;
+use crate::lotus::LotusClient;
+
+use super::subnet::SubnetManager;
+
+type MiddlewareImpl = SignerMiddleware<Provider<Http>, Wallet<SigningKey>>;
+
+pub struct EthSubnetManager<M: Send + Sync> {
```
Comment (cryptoAtwill) [inline src/manager/evm.rs:38]: I think we can exclude Send Sync in the struct as we are confining it in the impl

File: src/server/handlers/manager/subnet.rs [hunk 2]
```diff
@@ -4,24 +4,26 @@
...
 use ipc_identity::Wallet;
 use ipc_sdk::subnet_id::SubnetID;
 use std::sync::{Arc, RwLock};
 
 /// The subnet manager connection that holds the subnet config and the manager instance.
-pub struct Connection<T: JsonRpcClient> {
+pub struct Connection {
     subnet: Subnet,
-    manager: LotusSubnetManager<T>,
+    manager: Box<dyn SubnetManager>,
```
Comment (cryptoAtwill) [inline src/server/handlers/manager/subnet.rs:15]: Why not pass SubnetManager as generic parameter but use Box + dyn?

**Review by cryptoAtwill — APPROVED** [review]

File: src/manager/evm.rs (hunk 1 above)
Comment (adlrocha) [reply src/manager/evm.rs:38]: Good point, fixed!

File: src/server/handlers/manager/subnet.rs (hunk 2 above)
Comment (adlrocha) [reply src/server/handlers/manager/subnet.rs:15]: Oh, I forgot to push the version where I was using generics. That works, but good catch, thanks!

**Pull request closed without merging**